eclair-core/src/main/scala/fr/acinq/eclair/io/OpenChannelInterceptor.scala

```diff
@@ -0,0 +1,276 @@
+      }
+    }
+
+  def makeChannelParams(nodeParams: NodeParams, initFeatures: Features[InitFeature], upfrontShutdownScript_opt: Option[ByteVector], walletStaticPaymentBasepoint_opt: Option[PublicKey], isInitiator: Boolean, dualFunded: Boolean, fundingAmount: Satoshi, unlimitedMaxHtlcValueInFlight: Boolean): LocalParams = {
+    val maxHtlcValueInFlightMsat = if (unlimitedMaxHtlcValueInFlight) {
+      // We don't want to impose limits on the amount in flight, typically to allow fully emptying the channel.
+      21e6.btc.toMilliSatoshi
+    } else {
+      // NB: when we're the initiator, we don't know yet if the remote peer will contribute to the funding amount, so
+      // the percentage-based value may be underestimated. That's ok, this is a security parameter so it makes sense to
+      // base it on the amount that we're contributing instead of the total funding amount.
+      nodeParams.channelConf.maxHtlcValueInFlightMsat.min(fundingAmount * nodeParams.channelConf.maxHtlcValueInFlightPercent / 100)
+    }
+    LocalParams(
+      nodeParams.nodeId,
+      nodeParams.channelKeyManager.newFundingKeyPath(isInitiator), // we make sure that initiator and non-initiator key paths end differently
+      dustLimit = nodeParams.channelConf.dustLimit,
+      maxHtlcValueInFlightMsat = maxHtlcValueInFlightMsat,
+      requestedChannelReserve_opt = if (dualFunded) None else Some((fundingAmount * nodeParams.channelConf.reserveToFundingRatio).max(nodeParams.channelConf.dustLimit)), // BOLT #2: make sure that our reserve is above our dust limit
+      htlcMinimum = nodeParams.channelConf.htlcMinimum,
+      toSelfDelay = nodeParams.channelConf.toRemoteDelay, // we choose their delay
+      maxAcceptedHtlcs = nodeParams.channelConf.maxAcceptedHtlcs,
+      isInitiator = isInitiator,
+      upfrontShutdownScript_opt = upfrontShutdownScript_opt,
+      walletStaticPaymentBasepoint = walletStaticPaymentBasepoint_opt,
+      initFeatures = initFeatures
+    )
+  }
+
+}
+
+private class OpenChannelInterceptor(peer: ActorRef[Any],
+                                     pendingChannelsRateLimiter: ActorRef[PendingChannelsRateLimiter.Command],
+                                     pluginTimeout: FiniteDuration,
+                                     nodeParams: NodeParams,
+                                     wallet: OnchainPubkeyCache,
+                                     context: ActorContext[OpenChannelInterceptor.Command]) {
+
+  import OpenChannelInterceptor._
+
+  private def waitForRequest(): Behavior[Command] = {
+    receiveCommandMessage[WaitForRequestCommands](context, "waitForRequest") {
+      case initiator: OpenChannelInitiator => sanityCheckInitiator(initiator)
+      case nonInitiator: OpenChannelNonInitiator => checkRateLimits(nonInitiator)
+    }
+  }
+
+  private def sanityCheckInitiator(initiator: OpenChannelInitiator): Behavior[Command] = {
```

Comment:
nit: it's a bit confusing that you call this variable `initiator` (and in the other case `nonInitiator`), that's not the important meaning behind the variable, it's rather the `OpenChannel` part that is the most important...it makes more sense to call these variables `request` or `cmd` in both cases in my opinion.

Reply:
Fixed in [a032b09](https://github.com/ACINQ/eclair/pull/2552/commits/a032b097a5c1ccb8e04985f7714cc6fbd2937663).